### ISSUE TITLE
frequently reconnection problem

### ISSUE THREAD

**Issue by windfall617** [Victrid/splatplost#30]
While use the splatplost in splatoon3,meets the above obstacle, print every 4pit will eject the reconnection page= =
after 6 mins the splatplost stop  and never connect to the NS. How should  I do to deal with this problem?
PS this happened in the tv mode,because the oled screen....I am really worry about it's life....

**Comment by Victrid** [comment]
I thought the program is not ready for Splatoon 3 yet... I'm now running tests on TV mode for the development branch to support Splatoon 3. I think TV should not affect the printing as it's using bluetooth.
But your description is strange though. 
> print every 4pit will eject the reconnection page= =

What does this line mean? 4 pit means 4 dots or something? Can you share your running environment so we can dig into your problem?

**Comment by windfall617** [comment]
4 dots
the splatplot print every 4 dots will enter the reconnection page.
My environment is debian 20  on the Oracle VM virtual box.
And I also meet the problem with #28  XD

**Comment by Victrid** [comment]
Hello,
**For behaviour of VM**: It's recommended to use a physical linux machine. If you don't have one, you may check the [flashable image](https://github.com/Victrid/splatplost/blob/master/docs/image.md). Windows or macOS is not supported, as libnxctrl is based on bluez, the linux bluetooth stack.

**For splatoon 3**: I've updated a beta version which supports splatoon 3 and have a GUI. Check https://github.com/Victrid/splatplost/tree/dev-v0.2 for readme file for more information.

If you want to try this version, you need to install with special pip command, which allows you to install a pre-release version:

```bash
sudo pip install splatplost==0.2.0.dev0
```

If you still have problem connecting it, please feel free to reopen this issue!

**Comment by DetectiveNaoya** [comment]
This is an issue i'm now facing. I'm using the flashable image provided for refence.

**Comment by windfall617** [comment]
Hi :D 
After reading the above suggestions, I change my device to the raspberry 4B+,but while intsall the splatplost,I get trouble with the dependency conflict :(
How should I do to deal with it 
Thank u very much :))))
![ID7_44Q65VSV0O{43LH{VO6](https://user-images.githubusercontent.com/88772139/189511079-8c08eb0f-f3d1-4b42-bdb1-b9abeb68729d.png)

**Comment by windfall617** [comment]
And I also use the image,but I can't open the gui 
(I have follow "How to install".md  )
=....=  :((((((
![KM 7@)NASN FWHQKP0 VUG](https://user-images.githubusercontent.com/88772139/189511292-4b949e36-37ed-4627-a6f3-6e45a51533cb.png)

**Comment by Victrid** [comment]
> And I also use the image,but I can't open the gui (I have follow "How to install".md ) =....= :(((((( ![KM 7@)NASN FWHQKP0 VUG](https://user-images.githubusercontent.com/88772139/189511292-4b949e36-37ed-4627-a6f3-6e45a51533cb.png)

Try restart the konsole to get PATH refreshed? which splatplost version are you installing? If you want to install 0.2.0.dev0 you need to specify it (==0.2.0.dev0), or it will install 0.1.x.

**Comment by Victrid** [comment]
> Hi :D
> After reading the above suggestions, I change my device to the raspberry 4B+,but while intsall the splatplost,I get trouble with the dependency conflict :(
> How should I do to deal with it
> Thank u very much :))))

Check if you have installed something which depends on scipy and conflicts with 1.9.1. Use `pip show scipy`

**Comment by windfall617** [comment]
I have install the nxbt and joycontrol before ,and my image is the debian
![SFY %RZ6BW6Q@6Y`)~Q`QF](https://user-images.githubusercontent.com/88772139/189511652-5bb07f82-6aa0-4970-b3c7-ea95e2eb7a32.png)

**Comment by Victrid** [comment]
> I have install the nxbt and joycontrol before ,and my image is the debian

That is strange. 

> Thank u very much :))))
> ![ID7_44Q65VSV0O{43LH{VO6](https://user-images.githubusercontent.com/88772139/189511079-8c08eb0f-f3d1-4b42-bdb1-b9abeb68729d.png)

I see from this picture that you are trying to install a stable version (0.1.x) and not specified version number. splatplost GUI is only introduced in the new development version. Try:
```bash
sudo pip uninstall splatplost
```
to remove previous versions, and
```bash
sudo pip install splatplost==0.2.0.dev0
```
Maybe your raspberry pi have installed some package which causes conflict. You may try this with the image.

**Comment by lgorinator** [comment]
Yeah same problem here. I'm trying it on a fresh raspberry pi zero w raspbian. Tried to check the scipy version and got the scipy not found as well. I tried to install scipy by its self to resolve the issue, the highest successful version I have been able to install is 1.8.1

**Comment by DetectiveNaoya** [comment]
> Do you know what could cause the program to frequently disconnect with a Switch? For extra details I'm using the pre flashed image provided. When running the program I'm able to connect for the first time but it'd send me to the reconnection screen every time I tried to do anything with it, especially when it comes to the drawing process.

Didn't realize I posted this in the wrong issue, but the console doesn't really spit out any specific errors so I'm unsure as to what the cause of this could be

**Comment by Victrid** [comment]
> > Do you know what could cause the program to frequently disconnect with a Switch? For extra details I'm using the pre flashed image provided. When running the program I'm able to connect for the first time but it'd send me to the reconnection screen every time I tried to do anything with it, especially when it comes to the drawing process.
> 
> Didn't realize I posted this in the wrong issue, but the console doesn't really spit out any specific errors so I'm unsure as to what the cause of this could be

@DetectiveNaoya After using the nxbt as backend, I don't think the console will spit out any error regarding this. What's your bluetooth connector? You may use `lsusb` or `lspci` and find any entry with `Bluetooth` in it.

My bluetooth card will have multiple disconnection and connection before switch really connects, but it will not go to the drawing process. (it will wait on `open pair menu` on v0.1.x or `pairing` on the beta versions.) What is your situation?

**Comment by Yornas** [comment]
The entry I have is `IMC Networks Bluetooth Radio` and what happens for me is that it would just disconnect 5 seconds after `Press A` is clicked. Mine doesn't wait at `pairing` unlike yours.
